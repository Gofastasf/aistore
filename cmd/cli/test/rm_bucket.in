# Test individual bucket removal
ais bucket create ais://$BUCKET_1 | awk '{print $1 " " $2}'
ais bucket rm ais://$BUCKET_1

# Test multiple bucket removal
ais bucket create ais://$BUCKET_1 | awk '{print $1 " " $2}'
ais bucket create ais://$BUCKET_2 | awk '{print $1 " " $2}'
ais rmb ais://$BUCKET_1 ais://$BUCKET_2

# Test the --all feature: remove all AIS buckets
ais bucket create ais://$BUCKET_1 | awk '{print $1 " " $2}'
ais bucket create ais://$BUCKET_2 | awk '{print $1 " " $2}'
echo "DELETE" | ais rmb --all --yes

# Test when no AIS buckets exist
ais rmb --all --yes

# Test --all with incorrect confirmation response
ais bucket create ais://$BUCKET_1 | awk '{print $1 " " $2}'
ais bucket create ais://$BUCKET_2 | awk '{print $1 " " $2}'
echo "ILL-RESPONSE" | ais rmb --all --yes

# Test error cases
ais rmb ais://$BUCKET_1 --all // FAIL "cannot specify bucket name(s) with --all flag"

# Test invalid bucket URI
ais bucket rm hdfs://tmp // FAIL invalid
