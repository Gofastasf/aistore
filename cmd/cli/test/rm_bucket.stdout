"ais://$BUCKET_1" created
"ais://$BUCKET_1" destroyed
"ais://$BUCKET_1" created
"ais://$BUCKET_2" created
"ais://$BUCKET_1" destroyed
"ais://$BUCKET_2" destroyed
"ais://$BUCKET_1" created
"ais://$BUCKET_2" created
Found 2 AIS bucket(s) to remove:
  - ais://$BUCKET_1
  - ais://$BUCKET_2
Type 'DELETE' to confirm (or 'N' to cancel): "ais://$BUCKET_1" destroyed
"ais://$BUCKET_2" destroyed
No AIS buckets to remove
"ais://$BUCKET_1" created
"ais://$BUCKET_2" created
Found 2 AIS bucket(s) to remove:
  - ais://$BUCKET_1
  - ais://$BUCKET_2
Type 'DELETE' to confirm (or 'N' to cancel): Type 'DELETE' to confirm (or 'N' to cancel): Operation canceled.